### PR TITLE
Fixes #3231 adds developer documentation for journalist gui

### LINKED ISSUE #3231: [qt-journalist-updater] Add developer documentation
## Description

[See epic #3076 for primary feature, PRs for this ticket should go into `qt-journalist-updater`]

Add developer documentation for the journalist updater GUI

## User Stories

As a SecureDrop developer, I want documentation on how to develop the GUI such that I can contribute.

### PR DESCRIPTION
**Pull request by kushaldas**
## Status

Ready for review

## Description of Changes

Fixes #3231 We now have docs :)

Also added the `Pipfile` to create and maintain the virtual environment for the development.

## Testing

```
$ pip3 install pipenv --user
$ pipenv install
$ pipenv shell
```

## Deployment

Nothing now.

### If you made changes to documentation:

- [x] Doc linting (`make docs-lint`) passed locally

### CODE REVIEW

File: docs/development/updategui_development.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+
+::
+
+    $ pyrcc5 journalist_gui/resources.qrc -o journalist_gui/resources_rc.py
+
+
+
+.. note:: The *updaterUI.py* and *resources_rc.py* files are generated by the tools. So, do not make any changes to these
+          files. Any changes made to these files will be overridden.
```
[the remark this answers is not in the document]
Comment (kushaldas) [reply docs/development/updategui_development.rst]: Yup, I had that in mind, and forgot :(

File: docs/development/updategui_development.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+using Python3.
+
+Installing the dependencies in a virtual environment
+-----------------------------------------------------
+
+You can use `pipenv tool <https://docs.pipenv.org>`_ to create a virtualenv and install the dependencies.
+
+::
+
+    $ pip3 install pipenv --user
```
[the remark this answers is not in the document]
Comment (kushaldas) [reply docs/development/updategui_development.rst]: Yup, I thought let me do it and then ask :) Will fix that.

File: docs/development/updategui_development.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+The first command will create the virtualenv and install the dependencies. The second command is used to
+enable the Pipenv shell.
+
+
+To update the UI design
+-----------------------
+
+The design of the GUI is saved in the *journalist_gui/mainwindow.ui* file. To update the UI, one has to first install
```
[the remark this answers is not in the document]
Comment (kushaldas) [reply docs/development/updategui_development.rst]: Yup, I will update accordingly. Thanks.

File: docs/development/updategui_development.rst [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+
+.. note:: The ``updaterUI.py`` and ``resources_rc.py`` files are generated by the tools. So, do not make any changes to these
+          files. Any changes made to these files will be overridden.
+
+
+Adding and running test cases
+------------------------------
+
+We have Python unittests in the ``test_gui.py`` file. Any change in the actual application code will also require adding new
```
Comment (redshiftzero) [inline docs/development/updategui_development.rst]: Nit: `unittests` -> `unit tests`

**Comment by redshiftzero** [comment]
By the way, a really nice feature of our CI is that you can avoid running the long staging CI job by naming docs only branches `docs-*`

**Pull request merged**